Fix package namespace conflicts

Konamiman opened the pull request:
Moved all of the composer binary commands into the `composer-bin-plugin` to avoid version compatibility errors.

Context (from @ObliviousHarmony [in Slack](https://a8c.slack.com/archives/C0E1AV8T0/p1597812309064100)):

> I also made progress in this branch for a proper namespacing solution using `PHP Scoper`. The utility works by parsing an AST and then altering namespaces with a prefix. The standard process for this package is that it parses and dumps whatever you’re wanting into a build folder. I noticed that it supports a `-o` option for a targeted directory and if the input is a single folder then it flattens the export. My thinking here is that we can have a `vendor-src` directory set as our `vendor-dir` and after running `composer install` (probably with `--no-scripts` to avoid the autoloader error) and then process that into our desired `vendor` directory. We’ll need to tinker with the whitelist feature to avoid namespacing certain packages (Automattic, WooCommerce, probably, we can test everything thoroughly)

> This will allow us to develop against the prefixed dependencies instead of having to use their autoloader plugin. Since we’re using the Jetpack Autoloader my thinking is that it’s for the best if we do this instead of trying to use the automatic conversion feature against our own codebase.

> This branch integrates the [composer bin plugin](https://github.com/bamarni/composer-bin-plugin) as well in order to deal with the problems we are starting to run into with dev dependencies. It works by having the commands in their own folder and then symlinking it to the main `vendor/bin` directory. It’s pretty neat and I’ve moved all of the commands into it that I think we should. We could probably make the constraints on the respective `composer.json` files tighter but this was a WIP.

WIP!

### All Submissions:

* [ ] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes # .

### How to test the changes in this Pull Request:

1.
2.
3.

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

<!-- Mark completed items with an [x] -->

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

ObliviousHarmony committed:
Moved all of the composer binary commands into the `composer-bin-plugin` to avoid version compatibility errors

Konamiman committed:
Add improved PHP Scoper and a script to do the scoping after install.

rodrigoprimo commented:
I removed the milestone since this PR is still in draft. Please add the milestone back once this PR is ready for review.

Konamiman committed:
Improvements/fixes in the PHP Scoper configuration

- Proper specification of source directory
- No need to use files whitelist
- Preserve vendor-src (and gitignore it)
- Change the references to League\Container components to
  the prefixed ones

Konamiman committed:
Merge branch 'master' into fix/package-namespace-conflicts

Konamiman commented:
Original intent of the branch (introduction of Composer Bin) superseded by https://github.com/woocommerce/woocommerce/pull/27753, PHP Scoper introduction project abandoned.
